backend/app/api/v1/endpoints/leave_balances.py

```
@@ -0,0 +1,89 @@
+    settings: Settings = Depends(get_settings),
+    # _: bool = Depends(require_any_permissions([Permission.VIEW_LEAVE_BALANCE, Permission.VIEW_OWN_LEAVE_BALANCE]))
+) -> List[LeaveBalanceOut]:
+    """Retrieve leave balances for a user."""
+    try:
+        if user_id <= 0:
+            raise ValidationError(detail="Invalid user_id")
+        request_id = get_request_id(request)
+        return await get_leave_balances_by_user_and_type(user_id, leave_type, current_user, db, settings, request_id)
+    except ValidationError as e:
+        logger.error(f"Validation error: {str(e)}", extra={"request_id": request_id, "user_id": current_user.user_id})
+        raise HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY, detail=str(e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)